cpp/include/utilities/cython.hpp

```diff
@@ -15,53 +15,113 @@
  */
 #pragma once
 
+#include <experimental/graph_view.hpp>
 #include <graph.hpp>
-// #include <experimental/graph_view.hpp>
 #include <raft/handle.hpp>
 
 namespace cugraph {
 namespace cython {
 
-// FIXME: use std::variant instead of a union if possible
-// FIXME: add both CSRView and graph_type_t objects for easier testing during
-//        the transition
-union graphUnion {
-  graphUnion() {}
-  GraphCSRView<int, int, float> GraphCSRViewFloat;
-  GraphCSRView<int, int, double> GraphCSRViewDouble;
-};
+enum class numberTypeEnum : int { intType, floatType, doubleType };
 
-enum weightTypeEnum { floatType = 0, doubleType = 1 };
+// FIXME: The GraphCSRView* types are not in use! Those are left in place in
```

Comment:
Could add this in a later PR if desired.

Do we want to add GraphCSCViewFloat, GraphCSCViewDouble, GraphCOOViewFloat and GraphCOOViewDouble for completeness?  These are legacy classes, but it would allow all of the cython code to be modified to the new technique for graph allocation regardless of whether it hs been converted to the new graph objects or not.

Perhaps we defer this until we decide it would be useful.

This *might* simplify the later migration of all algorithms to MNMG.

Reply:
sounds good, I'll start this and see how involved it becomes. If it's holding up the PR too much I may defer it to the next PR.

Reply:
I added these and a FIXME to state how we might use them in the function when we need them.